geo/src/algorithm/hausdorff_distance.rs

```diff
@@ -0,0 +1,194 @@
+use crate::GeoFloat;
+use crate::CoordsIter;
+use crate::algorithm::EuclideanDistance;
+use num_traits::Bounded; // used to have T as generic type in folding
+use geo_types::*;
+
+/// Determine the distance between two geometries using the [Hausdorff distance formula].
+///
+/// [haversine formula]: https://en.wikipedia.org/wiki/Hausdorff_distance
+pub trait HausdorffDistance<T, Rhs = Self> {
+    fn hausdorff_distance(&self, rhs: &Rhs) -> T;
+}
+
+// We can take advantage of the coords_iter() method for all geometries
+// to iterate across all combos of coords (infinum). Simplifies the 
+// implementations for all geometries.
+macro_rules! impl_hausdorff_distance_coord_iter {
+    ($from:ident, [$($to:ident),*]) => {
+        $(
+            impl<T> HausdorffDistance<T, $to<T>> for $from<T>
+            where
+                T: GeoFloat 
+            {
+                fn hausdorff_distance(&self, geom: &$to<T>) -> T {
+                    // calculate from A -> B 
+                    let hd1 = self
+                        .coords_iter()
+                        .map(|c| {
+                            geom
+                            .coords_iter()
+                            .map(|c2| {
+                                c.euclidean_distance(&c2)
```

Comment:
Can you clarify in the docs that this algorithm utilizes euclidean distance under the hood? In case someone might be expecting it to use haversine or something else